platform/windows/display_server_windows.cpp

```diff
@@ -1507,6 +1516,172 @@ Ref<Image> DisplayServerWindows::screen_get_image_rect(const Rect2i &p_rect) con
+bool DisplayServerWindows::screen_is_hdr_supported(int p_screen) const {
+	_THREAD_SAFE_METHOD_
+
+	p_screen = _get_screen_index(p_screen);
+#ifdef D3D12_ENABLED
+	DXGI_OUTPUT_DESC1 desc1;
+	if (!_get_screen_desc(p_screen, desc1)) {
+		return false;
+	}
+
+	return (desc1.ColorSpace == DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020);
+#else
+	return false;
+#endif // D3D12_ENABLED
+}
+
+float DisplayServerWindows::screen_get_min_luminance(int p_screen) const {
+	_THREAD_SAFE_METHOD_
+
+	p_screen = _get_screen_index(p_screen);
+#ifdef D3D12_ENABLED
+	DXGI_OUTPUT_DESC1 desc1;
+	if (!_get_screen_desc(p_screen, desc1)) {
+		return 0.0f;
+	}
+
+	return desc1.MinLuminance;
+#else
+	return 0.0f;
+#endif // D3D12_ENABLED
+}
+
+float DisplayServerWindows::screen_get_max_luminance(int p_screen) const {
+	_THREAD_SAFE_METHOD_
+
+	p_screen = _get_screen_index(p_screen);
+#ifdef D3D12_ENABLED
+	DXGI_OUTPUT_DESC1 desc1;
+	if (!_get_screen_desc(p_screen, desc1)) {
+		return 0.0f;
+	}
+
+	return desc1.MaxLuminance;
+#else
+	return 0.0f;
+#endif // D3D12_ENABLED
+}
+
+float DisplayServerWindows::screen_get_max_full_frame_luminance(int p_screen) const {
+	_THREAD_SAFE_METHOD_
+
+	p_screen = _get_screen_index(p_screen);
+#ifdef D3D12_ENABLED
+	DXGI_OUTPUT_DESC1 desc1;
+	if (!_get_screen_desc(p_screen, desc1)) {
+		return 0.0f;
+	}
+
+	return desc1.MaxFullFrameLuminance;
+#else
+	return 0.0f;
+#endif // D3D12_ENABLED
+}
+
+static BOOL CALLBACK _MonitorEnumProcSdrWhiteLevel(HMONITOR hMonitor, HDC hdcMonitor, LPRECT lprcMonitor, LPARAM dwData) {
+	EnumSdrWhiteLevelData *data = (EnumSdrWhiteLevelData *)dwData;
+	if (data->count == data->screen) {
+		MONITORINFOEXW minfo;
+		memset(&minfo, 0, sizeof(minfo));
+		minfo.cbSize = sizeof(minfo);
+		GetMonitorInfoW(hMonitor, &minfo);
+
+		// First, find this screen's path.
+		for (const DISPLAYCONFIG_PATH_INFO &path : data->paths) {
+			DISPLAYCONFIG_SOURCE_DEVICE_NAME source_name;
+			memset(&source_name, 0, sizeof(source_name));
+			source_name.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
+			source_name.header.size = sizeof(source_name);
+			source_name.header.adapterId = path.sourceInfo.adapterId;
+			source_name.header.id = path.sourceInfo.id;
+
+			if (DisplayConfigGetDeviceInfo(&source_name.header) == ERROR_SUCCESS) {
+				if (wcscmp(minfo.szDevice, source_name.viewGdiDeviceName) == 0) {
+					// Found it, now we can query the SDR white level.
+					DISPLAYCONFIG_SDR_WHITE_LEVEL sdr_white_level;
+					memset(&sdr_white_level, 0, sizeof(sdr_white_level));
+					sdr_white_level.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SDR_WHITE_LEVEL;
+					sdr_white_level.header.size = sizeof(sdr_white_level);
+					sdr_white_level.header.adapterId = path.targetInfo.adapterId;
+					sdr_white_level.header.id = path.targetInfo.id;
+
+					LONG result = DisplayConfigGetDeviceInfo(&sdr_white_level.header);
+					if (result == ERROR_SUCCESS) {
+						data->sdrWhiteLevelInNits = (float)(sdr_white_level.SDRWhiteLevel / 1000) * 80;
```

Comment:
```suggestion
						data->sdrWhiteLevelInNits = (float)sdr_white_level.SDRWhiteLevel / 1000 * 80;
```

This suggestion corrects reading of window SDR white level to match behaviour described in the documentation:
https://learn.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-displayconfig_sdr_white_level

(Current PR code incorrectly performs integer division instead of float division.)

Reply:
Good catch, fixed.